Mail/Views/Search/SearchView.swift

```diff
@@ -0,0 +1,270 @@
+
+    @AppStorage(UserDefaults.shared.key(.threadDensity)) var threadDensity = ThreadDensity.normal
+
+    @StateObject var bottomSheet: ThreadBottomSheet
+    @State private var navigationController: UINavigationController?
+
+    private let bottomSheetOptions = Constants.bottomSheetOptions + [.appleScrollBehavior]
+
+    let isCompact: Bool
+
+    init(mailboxManager: MailboxManager, folder: Folder?, isCompact: Bool) {
+        let threadBottomSheet = ThreadBottomSheet()
+        _bottomSheet = StateObject(wrappedValue: threadBottomSheet)
+        _viewModel = StateObject(wrappedValue: SearchViewModel(mailboxManager: mailboxManager, folder: folder))
+        self.isCompact = isCompact
+    }
+
+    var body: some View {
+        VStack {
+            ScrollView(.horizontal, showsIndicators: false) {
+                HStack(spacing: 0) {
+                    ForEach(viewModel.filters) { filter in
+                        if filter == .folder {
+                            SearchFilterFolderCell(selection: $viewModel.selectedSearchFolderId, folders: viewModel.folderList)
+                                .padding(.horizontal, 12)
+                        } else {
+                            SearchFilterCell(
+                                title: filter.title,
+                                isSelected: viewModel.selectedFilters.contains(filter)
+                            )
+                            .padding(.vertical, 2)
+                            .padding(.trailing, 0)
+                            .padding(.leading, 12)
+                            .onTapGesture {
+                                viewModel.updateSelection(filter: filter)
+                            }
+                        }
+                    }
+                }
+            }
+            .padding(.top, 32)
+
+            List {
+                if viewModel.showHistory {
+                    searchHistoryList(history: viewModel.searchHistory)
+                } else {
+                    if !viewModel.contacts.isEmpty {
+                        contactList(contacts: viewModel.contacts)
+                    }
+
+                    if !viewModel.threads.isEmpty {
+                        threadList(threads: viewModel.threads)
+                    } else if viewModel.searchInfo.hasSearched {
+                        // TODO: - Add no result view
+                        EmptyView()
+                    }
+
+                    if viewModel.searchInfo.isLoading {
+                        HStack {
+                            Spacer()
+                            ProgressView()
+                            Spacer()
+                        }
+                        .listRowSeparator(.hidden)
+                    }
+                }
+            }
+            .listStyle(.plain)
+        }
+        .introspectNavigationController { navigationController in
+            let newNavController = navigationController
```

Comment:
Why use newNavController ?
Also please extract code to helper like in kDrive (https://github.com/Infomaniak/ios-kDrive/blob/master/kDrive/Utils/UINavigationController%2BExtension.swift) as code is duplicated from ThreadView